### DISCUSSION TITLE
TypeError: Object(...)(...) is undefined and TypeError: t is undefined

### DISCUSSION THREAD

**Discussion by kylemiller3** [vercel/next.js#19986] [Help] [answered]
I am getting both these errors intermittently at the same time. For additional context I am using nextjs redux since it seems to be a problem with hydration. I do not understand why it happens only sometimes. Any ideas on how I can fix or debug this error?

```
TypeError: Object(...)(...) is undefined
    NextJS 41
        Fa
        qi
        yo
        su
        uu
        eu
        Yo
        Lu
        ju
        nu
        ju
        hydrate
        de
        e
        f
        _invoke
        m
        r
        s
        exports
        exports
        ae
        oe
        e
        f
        _invoke
        m
        r
        s
        promise callback*r
        s
        promise callback*r
        s
        exports
        exports
        ne
        BMP1
        u
        t
        r
        <anonymous>
framework.53cfa66f9846f4090096.js:1:94850
    NextJS 44
        eo
        callback
        di
        io
        mu
        unstable_runWithPriority
        jl
        pu
        eu
        Yo
        Lu
        ju
        nu
        ju
        hydrate
        de
        e
        f
        _invoke
        m
        r
        s
        exports
        exports
        ae
        oe
        e
        f
        _invoke
        m
        r
        s
        (Async: promise callback)
    r
        s
        (Async: promise callback)
    r
        s
        exports
        exports
        ne
        BMP1
        u
        t
        r
        <anonymous>
```
```
TypeError: t is undefined
    NextJS 41
        Oa
        qi
        yo
        su
        uu
        eu
        Yo
        Lu
        ju
        nu
        ju
        hydrate
        de
        e
        f
        _invoke
        m
        r
        s
        exports
        exports
        ae
        oe
        e
        f
        _invoke
        m
        r
        s
        promise callback*r
        s
        promise callback*r
        s
        exports
        exports
        ne
        BMP1
        u
        t
        r
        <anonymous>
framework.53cfa66f9846f4090096.js:1:94850
```

**Comment by jamesmosier** [comment, marked as answer]
In order to track this down, you'll need to find out what `t` is and why it is undefined. A quick way to do this would be to click on the link to the source code that is throwing the error in your browsers console. From there you'll likely see some obfuscated code and you'll need to try to look for what code is around it to determine where in your app it is. The other way would be to put a breakpoint in the source via your browsers debugger and that would help as well.

**Reply by kylemiller3** [reply]
Thank you for your response. I have been trying to reproduce this bug reliably and I believe this is a Firefox only bug as I cannot reproduce it in either Chromium nor Safari. The code around it does not look familiar at all as it's in the framework code. Not sure how to breakpoint it because it's all one line. Knowing all this, how likely is it a bug in NextJS or Firefox rather than my code?

![screenshot_20201208_225440](https://user-images.githubusercontent.com/5983525/101580363-6400e000-39a8-11eb-9cca-eaee0a386fb5.png)

![screenshot_20201208_225706](https://user-images.githubusercontent.com/5983525/101581554-b80bc480-39a8-11eb-981a-b1a70871ba48.png)

I have a live site where I have figured out how to reproduce it semi-reliably in Firefox but I'm really at a loss on how to debug this or if it's even my bug or worth the effort at all.

**Reply by jamesmosier** [reply]
Sorry, without seeing the site or the code unfortunately I won't be of much assistance.

**Reply by kylemiller3** [reply]
The problem was a react-redux wrapper I was using
